lib/shopify-cli/git.rb

```diff
@@ -121,19 +123,20 @@ def rev_parse(*args, dir: nil, ctx: Context.new)
       def clone_progress(*git_command, ctx:)
         CLI::UI::Progress.progress do |bar|
           msg = []
-          success = ctx.system('git', *git_command, '--progress') do |_out, err|
-            if err.strip.start_with?('Receiving objects:')
-              percent = (err.match(/Receiving objects:\s+(\d+)/)[1].to_f / 100).round(2)
+          success = Open3.popen3('git', *git_command, '--progress') do |_stdin, _stdout, stderr, thread|
```

Comment:
Any reason we don't just use `ctx.capture3(..)` or something? Might feel a little more inline with how we usually handle command execution

Reply:
`ctx.capture3` calls `Open3.capture3` and what Fred needs here is `Open3.popen3` to enable the block that is called here piping all input back into this ruby code.

Reply:
⬆️  What @tanema  said. My initial investigations did exactly as you say to try to use the `ctx.capture3` but it doesn't allow the use of the block syntax which is what allowed me to access the output pipe as it occurred. 

I am open to another solution if you know of a way to get a similar effect with the `ctx.capture3`.

Reply:
Cool, that makes sense then! Thanks for clarifying